src/server/game/Spells/SpellEffects.cpp

```diff
@@ -3020,6 +3020,13 @@ void Spell::EffectSummonPet(SpellEffIndex effIndex)
             //OldSummon->Relocate(px, py, pz, OldSummon->GetOrientation());
             //OldSummon->SetMap(owner->GetMap());
             //owner->GetMap()->Add(OldSummon->ToCreature());
+            if (OldSummon->getPetType() == SUMMON_PET)
+            {
+                OldSummon->SetHealth(OldSummon->GetMaxHealth());
+                OldSummon->SetPower(OldSummon->getPowerType(),
+                OldSummon->GetMaxPower(OldSummon->getPowerType()));
```

Comment:
indentation on this line is VERY unclear, it is easy to miss that this is a continuation of previous line